php/commands/core.php

```diff
@@ -134,7 +134,8 @@ public function download( $args, $assoc_args ) {
 		$locale = \WP_CLI\Utils\get_flag_value( $assoc_args, 'locale', 'en_US' );
 
 		if ( isset( $assoc_args['version'] ) ) {
-			$version = $assoc_args['version'];
+			$version = strtolower( $assoc_args['version'] );
+			$version = ( 'nightly' === $version ? 'trunk' : $version );
 			$download_url = $this->get_download_url($version, $locale, 'tar.gz');
```

Comment:
> So I think we should either note in the docbloc that get_download_url() always returns a zip for trunk version or call WP_CLI::error(). The second option doesn't feel necessary.

Ok. Just to make sure the execution path is accommodated for, let's call `WP_CLI::error()` when `tar.gz` is provided as a version for nightly.